core/src/scene/styleContext.cpp

```diff
@@ -313,14 +327,10 @@ bool StyleContext::evalFunction(FunctionID id) {
 
 bool StyleContext::evalFilter(FunctionID _id) {
 
-    bool result = false;
-
     if (!evalFunction(_id)) { return false; };
 
-    // check for evaluated value sitting at value stack top
-    if (duk_is_boolean(m_ctx, -1)) {
-        result = duk_get_boolean(m_ctx, -1);
-    }
+    // Evaluate the "truthiness" of the function result at the top of the stack.
+    bool result = duk_to_boolean(m_ctx, -1);
```

Comment:
The top of the stack at this point will be the return value of the function that was just evaluated. For a filter, we want to evaluate whether this value is "truthy" or not - since this is what tangram-JS evaluates.